source/2_platform/rex_windows/src/input/input.cpp

```diff
@@ -0,0 +1,72 @@
+#include "rex_windows/input/input.h"
+
+#include "rex_engine/diagnostics/win/win_call.h"
+#include "rex_std/algorithm.h"
+#include "rex_std/assert.h"
+#include "rex_std/vector.h"
+
+#define NOMINMAX
+#include <Windows.h>
+
+namespace rex
+{
+  namespace input
+  {
+    // NOLINTBEGIN(cppcoreguidelines-avoid-non-const-global-variables)
+    card32 g_current_input_idx               = 1;
+    rsl::array<bool, 256> g_keys_even_frames = {};
+    rsl::array<bool, 256> g_keys_odd_frames  = {};
+    rsl::array<bool, 256>* g_prev_keys       = &g_keys_even_frames;
+    rsl::array<bool, 256>* g_current_keys    = &g_keys_odd_frames;
```

Comment:
Why are these globals?

Reply:
why shouldn't they be?

Reply:
allocated memory in global scope instead of using a function for it and returning a static.

Reply:
what is allocated?

Reply:
there's no dynamic allocated memory here

Reply:
No, but you still make room for 256 bytes (* 4) while if there is no input this is unused memory. Aren't you the one who mentioned keeping a low memory profile?

Reply:
it doesn't matter if they're in a function or not, the memory will be there anyway, it'll be in the exe data sector

Reply:
It loads the memory already even within a function?

Reply:
yes, it's static memory, so it'll always be there

Reply:
But if the function is not called how does it know it needs to initialize this?

Reply:
initialization is different from memory allocation.
the memory is allocated but the lifetime of the object isn't started until the function is called for the first time

Reply:
Where have you seen this?
Cuz I've googled it and I cannot find a reference, they only talk about initialization.

Reply:
https://stackoverflow.com/questions/93039/where-are-static-variables-stored-in-c-and-c

Reply:
Oke I understand now, there is no use in moving this.